src/Function.ts

```diff
@@ -0,0 +1,131 @@
+import {
+  AllMiddlewareArgs,
+  AnyMiddlewareArgs,
+  Middleware,
+  SlackEventMiddlewareArgs,
+} from './types';
+
+/* Types */
+export interface SlackFunctionExecutedMiddlewareArgs extends SlackEventMiddlewareArgs<'function_executed'> {
+  success: SuccessFn;
+  error: ErrorFn;
+}
+
+export type AllSlackFunctionExecutedMiddlewareArgs = SlackFunctionExecutedMiddlewareArgs & AllMiddlewareArgs;
+
+/**
+ * A Function is a deterministic machine with
+ * specified outputs given specific inputs.
+ * --
+ * You configure a Function's title, inputs, and outputs
+ * in your project's manifest.json. If your project contains any
+ * functions via app.function, it must have a corresponding
+ * manifest.json entries or App will throw an error when attempting to
+ * initialize.
+ * --
+ * Slack will take care of providing inputs to your function
+ * via a function_execution event. Bolt handles delivering those
+ * to your function in the way you can expect of regular events,
+ * messages, shortcuts commands, etc.
+ * --
+ * When initiating an instance of Function below, you supply the
+ * fn you want to process the supplied inputs and what logical
+ * conditions determine success or failure in your use case.
+ * You must call the supplied utility success with your specified
+ * outputs or failure.
+ * */
+export class Function {
```

Comment:
I thought about naming this `SlackFunction`

Reply:
Yeah I would refrain from using Function if only because it is a [standard built-in object in JS](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function), which could be confusing. I am surprised this compiles tbh! I figured the interpreter would complain.